### ISSUE TITLE
Yolo12 pretrain

### ISSUE THREAD

**Issue by czheyutao** [ultralytics/ultralytics#21046]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I encountered a problem reproducing YOLO12 on the Coco dataset.Why do the effects obtained by val in these two situations not match.In addition, I was unable to find the training settings through model.ckpt in yolo12.pt

![Image](https://github.com/user-attachments/assets/2a3be4b4-7f86-4d61-a104-00a8c3103846)

![Image](https://github.com/user-attachments/assets/4a3d048e-fe51-4ea5-b280-e35da5a10950)

![Image](https://github.com/user-attachments/assets/d838c048-79db-46a2-8521-234d42bdb437)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @czheyutao, thank you for your interest in Ultralytics and for trying out YOLO12 on the COCO dataset! 🚀 This is an automated response to help guide you while an Ultralytics engineer reviews your issue and will assist you soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) where you can find helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples. Many common questions about training, validation, and pretrained weights are addressed there.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug the issue. This should include the exact commands used, the dataset configuration, and any environment details that could help us reproduce your results.

If this is a custom training ❓ Question, please include as much information as possible—training logs, dataset samples, and verify you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

For real-time community chat, join us on [Discord](https://discord.com/invite/ultralytics) 🎧. For in-depth discussions, visit [Discourse](https://community.ultralytics.com/), or connect on the [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Ensure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date, verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for your patience and for providing any additional details requested above. An Ultralytics engineer will be with you soon!

**Comment by Y-T-G** [comment]
How did you get the first one?

**Comment by czheyutao** [comment]
> How did you get the first one?
Sorry for the vague expression. I use "model = YOLO("yolo12n.yaml").load("yolo12n.pt")" to get the first result

**Comment by czheyutao** [comment]
> How did you get the first one?

Sorry for the vague expression.I use the code below to get the first result

![Image](https://github.com/user-attachments/assets/85cb8353-fcf6-45e4-a4d1-e79c233fc792)

**Comment by Y-T-G** [comment]
@Laughing-q Can you check this? I can reproduce it. Loading to YAML gets lower mAP compared to loading the model directly even though all weights are transferred.

```
model = YOLO("yolo12n.yaml").load("yolo12n.pt") # lower mAP
model = YOLO("yolo12n.pt")  # correct mAP
```

**Comment by Laughing-q** [comment]
@Y-T-G @czheyutao Hey I debugged a bit and found it's related to official yolov12-1.0 weights.
The root here is that the `bias=True` in [yolo12 AAttn.pe.conv](https://github.com/ultralytics/ultralytics/blob/b6824625659ba12c05b349d3ca8a1d5c707e38c9/ultralytics/nn/modules/block.py#L1733-L1733) module, but in ultralytics package we have `bias=False` all the time.
https://github.com/ultralytics/ultralytics/blob/b6824625659ba12c05b349d3ca8a1d5c707e38c9/ultralytics/nn/modules/conv.py#L66-L66
- Directly loading `yolo12n.pt` would load the original weights and all the module settings exactly the same as it was trained, hence despite we have `bias=False` in `Conv` module it'd also load as `bias=True`.
- Creating model by `yolo12n.yaml` and load weights afterwards would stick with `bias=False`.

My guess is that the yolo12 team set bias=True for R&D experiments hence the 1.0 weights are trained with `bias=True` here. And then somehow reset it back to False.
Reference: 
- https://github.com/sunsmarterjie/yolov12/commit/0aa42779bf7048b418e04c081ed9a86b7cc79a53
- https://github.com/sunsmarterjie/yolov12/commit/acd0ee670c9303535b161234c07a8ea72fdba4f8

Reproducible example using official yolov12 repo:
```bash
git clone https://github.com/sunsmarterjie/yolov12
cd yolov12
git checkout 0aa42779bf7048b418e04c081ed9a86b7cc79a53
pip install -e.
```
```python
def test_yolo12():
    # need the weights from yolo12-1.0
    # https://github.com/sunsmarterjie/yolov12/releases/download/v1.0/yolov12n.pt
    from ultralytics import YOLO

    model = YOLO("yolov12n.pt")
    with open("yolo12n-pt.txt", "w") as f:
        f.write(model.__str__())
    model = YOLO("yolov12n.yaml").load("yolov12n.pt") # lower mAP
    with open("yolo12n-load.txt", "w") as f:
        f.write(model.__str__())
    model.val(data="coco8.yaml", verbose=False)

test_yolo12()
```
![Image](https://github.com/user-attachments/assets/0d4a8995-ce9b-4f1f-9fd2-7f67e415a486)
Also you can get the saved `yolo12n-pt.txt` and `yolo12n-load.txt` to checkout the `bias=False/True` difference there.

**Comment by Laughing-q** [comment]
@czheyutao based on the results above we couldn't really do something here, I would recommend always using `YOLO("yolo12.pt")` for all the use cases to avoid this issue.

**Comment by czheyutao** [comment]
Thank you very much for your answer. The reason why I didn't use the official yolo12 is that my device doesn't support flashattention. Ultralytics provides an implementation that does not need it. I would like to ask whether the same accuracy can be achieved by using ultralytics yolo12n.yaml instead of using pre training weights for coco reproduction. This is what I am trying to do

**Comment by glenn-jocher** [comment]
@czheyutao Training from scratch with `yolo12n.yaml` on COCO should theoretically achieve similar accuracy to the pretrained weights, but YOLO12 is not recommended due to training instability, excessive memory consumption, and 2-3x slower CPU performance compared to YOLO11. I'd strongly recommend using [YOLO11](https://docs.ultralytics.com/models/yolo11/) instead, which provides better stability, performance, and doesn't require FlashAttention while delivering state-of-the-art results across all tasks.
